lighthouse-core/test/create-test-trace.js

```diff
@@ -57,14 +57,14 @@ function getChildTask({ts, duration, url}) {
  * Creates a simple trace that fits the desired options. Useful for basic trace
  * generation, e.g a trace that will result in particular long-task quiet
  * periods. Input times should be in milliseconds.
- * @param {{navigationStart?: number, traceEnd?: number, topLevelTasks?: Array<TopLevelTaskDef>}} options
```

Comment:
Hm, not sure I agree here. The primary goal after this change is to have all the places where we say `navigationStart` do so because conceptually they absolutely require a `navigationStart` event and not just a `timeOrigin` timestamp.

All the places where we use createTestTrace today don't need absolutely require a navigationStart event (other than the fact that it's an implementation detail of determining our timeOrigin at this moment), they're just trying to set the timeOrigin. If and when we have tests that explicitly require a navigationStart event because it's the only thing that allows the computation to work, then I agree we should reintroduce `navigationStart` as a secondary option in this object.

I guess what I'm saying is, everything that sets `navigationStart` in `createTestTrace` today would be fine if we replaced the `navigationStart` event with say `FetchStart` if it were the `timeOriginEvt` in TraceProcessor.

Does that make sense?

Reply:
> Does that make sense?

I still disagree, but too late now :P

> I guess what I'm saying is, everything that sets navigationStart in createTestTrace today would be fine if we replaced the navigationStart event with say FetchStart if it were the timeOriginEvt in TraceProcessor.

but not a user timing event.

We don't have a definition for TTI for an arbitrary starting point since it requires FCP and DCL. By removing specificity we make every instance of `timeOrigin` into a hunt into other files for where the trace came from and/or what starting point we were interested in (instead of just saying, oh, yeah, navStart and moving on in the code). Instead I think it's better to stay specific whenever possible (like in test code for tests explicitly written based on navStart) and only get general when we need to be (e.g. in the case of `create-test-trace`, `navigationStart` could have been made optional and other possible origins added, with the generated trace adapted depending on which one is used)